src/main/java/beam/analysis/plots/RealizedModeStats.java

```diff
@@ -0,0 +1,309 @@
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static beam.sim.metrics.Metrics.ShortLevel;
+
+public class RealizedModeStats implements IGraphStats, MetricsSupport {
+
+
+    private static Map<Integer, Map<String, Integer>> hourModeFrequency = new HashMap<>();
+    private static final String graphTitle = "Realized Mode Histogram";
+    private static final String xAxisTitle = "Hour";
+    private static final String yAxisTitle = "# mode chosen";
+    private static final String fileName = "realized_mode";
+    private static List<String> personIdList = new ArrayList<>();
+    private Logger log = LoggerFactory.getLogger(this.getClass());
+    private static Map<ModePerson, Integer> hourPerson = new HashMap<>();
+
+    @Override
+    public void processStats(Event event) {
+        processRealizedMode(event);
+    }
+
+    @Override
+    public void createGraph(IterationEndsEvent event) throws IOException {
+
+        Map<String, String> tags = new HashMap<>();
+        tags.put("stats-type", "aggregated-mode-choice");
+        hourModeFrequency.values().stream().flatMap(x -> x.entrySet().stream())
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, (a, b) -> a + b))
+                .forEach((mode, count) -> countOccurrenceJava(mode, count, ShortLevel(), tags));
+
+        CategoryDataset modesFrequencyDataset = buildModesFrequencyDatasetForGraph();
+        if (modesFrequencyDataset != null)
+            createModesFrequencyGraph(modesFrequencyDataset, event.getIteration());
+
+        writeToCSV(event);
+    }
+
+    @Override
+    public void createGraph(IterationEndsEvent event, String graphType) throws IOException {
+
+    }
+
+    @Override
+    public void resetStats() {
+        hourModeFrequency.clear();
+        personIdList.clear();
+        hourPerson.clear();
+    }
+
+
+    private void processRealizedMode(Event event) {
+        int hour = GraphsStatsAgentSimEventsListener.getEventHour(event.getTime());
+        Map<String, Integer> hourData = hourModeFrequency.get(hour);
+        if (event.getEventType() == ModeChoiceEvent.EVENT_TYPE) {
+
+            String mode = event.getAttributes().get(ModeChoiceEvent.ATTRIBUTE_MODE);
+            String personId = event.getAttributes().get(ModeChoiceEvent.ATTRIBUTE_PERSON_ID);
+            Map<String, String> tags = new HashMap<>();
+            tags.put("stats-type", "mode-choice");
+            tags.put("hour", "" + (hour + 1));
+            countOccurrenceJava(mode, 1, ShortLevel(), tags);
+            if (personIdList.contains(personId)) {
+                personIdList.remove(personId);
+                return;
+            }
+
+            Integer frequency = 1;
+            if (hourData != null) {
+                frequency = hourData.get(mode);
+                if (frequency != null) {
```

Comment:
⚠️ Simplify expression to make the code less verbose and increase readability/maintainability.

## Suggestion

Before:
```Java
frequency = hourData.get(mode);
if (frequency != null) {
    frequency++;
} else {
    frequency = 1;
}
```

After:
```Java
frequency = hourData.getOrDefault(mode, 0) + 1;
```